src/components/Transfer.tsx

```diff
@@ -47,54 +50,51 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 function Transfer() {
+  const { dispatchTransaction } = useUpdateTransactionContext();
   const classes = useStyles();
   const [isRunning, setIsRunning] = useState(false);
   const [helperText, setHelperText] = useState('');
-  const [transferInput, setTransferInput] = useState<string>('');
-  const [actualInput, setActualInput] = useState<number | null>();
   const [amountNotCorrect, setAmountNotCorrect] = useState<boolean>(false);
   const { sourceChainDetails, targetChainDetails } = useSourceTarget();
   const { account } = useAccounts();
 
-  const planck = 10 ** targetChainDetails.apiConnection.api.registry.chainDecimals[0];
-  const { estimatedFee, receiverAddress, estimatedFeeLoading } = useTransactionContext();
+  const { estimatedFee, transferAmount, receiverAddress, estimatedFeeLoading } = useTransactionContext();
   const { api, isApiReady } = sourceChainDetails.apiConnection;
   const balance = useBalance(api, account?.address || '');
 
   const { isButtonDisabled, sendLaneMessage } = useSendMessage({
-    input: actualInput?.toString() ?? '',
+    input: transferAmount ?? '',
     isRunning,
     setIsRunning,
     type: TransactionTypes.TRANSFER
   });
 
   const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (event.target.value) {
-      const [actualValue, message] = evalUnits(event.target.value);
-      setHelperText(message);
-      setActualInput(actualValue && actualValue * planck);
-    }
-    setTransferInput(event.target.value);
+    const [actualValue, message] = evalUnits(event.target.value);
+    setHelperText(message);
+    dispatchTransaction(
+      TransactionActionCreators.setTransferAmount(null, actualValue !== null ? actualValue?.toString() : '')
+    );
   };
 
   useEffect((): void => {
-    isRunning && setTransferInput('');
-  }, [isRunning]);
+    isRunning && dispatchTransaction(TransactionActionCreators.setTransferAmount(null, ''));
+  }, [dispatchTransaction, isRunning]);
 
   // To extract estimated fee logic to specific component. Issue #171
   useEffect((): void => {
     estimatedFee &&
-      actualInput &&
-      setAmountNotCorrect(new BN(balance.free).sub(new BN(actualInput).add(new BN(estimatedFee))).toNumber() < 0);
-  }, [actualInput, estimatedFee, balance, isApiReady]);
+      transferAmount &&
+      setAmountNotCorrect(new BN(balance.free).sub(new BN(transferAmount).add(new BN(estimatedFee))).isNeg());
+  }, [transferAmount, estimatedFee, balance, isApiReady]);
```

Comment:
Since we have now "smarter" reducers that handle more complex logic, this useEffect could be removed and let the `setTransferAmount` action to take care of it. Then you can use `transferAmountError` to set whatever error message you would like to share with the user.

Reply:
See above.